### PR TITLE
[rllib] Expose PPO evaluator resource requirements

### PR DESCRIPTION
**Pull request by pcmoritz**
## What do these changes do?

This makes it possible for users to provide custom resources for the PPO evaluator. This can for example be used to run simulators that require GPUs to run.

### CODE REVIEW

File: python/ray/rllib/ppo/ppo.py [hunk 1]
```diff
@@ -67,6 +67,8 @@
     "min_steps_per_task": 1000,
     # Number of actors used to collect the rollouts
     "num_workers": 5,
+    # Resource requirements for simulation actors
+    "worker_resources": {"num_cpus": 1},
```
Comment (richardliaw) [inline python/ray/rllib/ppo/ppo.py]: `evaluator_resources`

Comment (richardliaw) [reply python/ray/rllib/ppo/ppo.py]: @ericl?

File: python/ray/rllib/ppo/ppo.py [hunk 2]
```diff
@@ -91,6 +93,7 @@ def _init(self):
         self.kl_coeff = self.config["kl_coeff"]
```
Comment (richardliaw) [inline python/ray/rllib/ppo/ppo.py]: You may want to modify L88 (`  _allow_unknown_subkeys = ["model", "tf_session_args"]`) to include `evaluator_resources`

**Review by richardliaw — CHANGES_REQUESTED** [review]

**Comment by richardliaw** [comment]
Overall this is fine; I think we should be careful referring to the remote evaluators rather than agents, actors, workers, etc

File: python/ray/rllib/ppo/ppo.py (hunk 1 above)
Comment (ericl) [reply python/ray/rllib/ppo/ppo.py]: worker sounds good to me

File: python/ray/rllib/ppo/ppo.py (hunk 2 above)
Comment (ericl) [reply python/ray/rllib/ppo/ppo.py]: Yep or alternatively provide gpu:0 there

**Comment by pcmoritz** [comment]
@richardliaw This is ready to merge, the test failure is unrelated

**Review by richardliaw — APPROVED** [review]

**Pull request merged**